src/js/options/pages/preferences/index.js

```diff
@@ -42,14 +42,31 @@ function Prefs() {
     [setPreference]
   );
 
+  const requestPermission = useCallback(
+    () =>
+      browser.permissions.request(
+        {
+          permissions: [],
+          origins: ['<all_urls>']
+        },
+        granted => {
+          // The callback argument will be true if the user granted the permissions.
+          if (granted) {
+            onChange('alertOnSubmit', !alertOnSubmit);
+          }
+        }
+      ),
+    [alertOnSubmit, onChange]
+  );
+
   return (
     <>
       <form>
         <div className={styles.pageSection}>
           <FormCheckbox
             name="alertOnSubmit"
             checked={alertOnSubmit}
-            onChange={() => onChange('alertOnSubmit', !alertOnSubmit)}
+            onChange={() => requestPermission()}
```

Comment:
can we revoke permission if they uncheck it again?

Reply:
There's no way to revoke permissions AFAICT